src/modm/driver/display/ssd1306_i2c_impl.hpp

```diff
@@ -0,0 +1,138 @@
+
+	RF_CALL(initializeMemoryMode());
+
+	this->commandBuffer[0] = uint8_t(ssd1306::TimingAndDrivingCommands::ChargePump);
+	this->commandBuffer[1] = uint8_t(ssd1306::ChargePump::V7_5);
+	this->commandBuffer[2] = uint8_t(ssd1306::HardwareConfigCommands::SegmentRemap127);
+	this->commandBuffer[3] = uint8_t(ssd1306::HardwareConfigCommands::ComOutputScanDirectionDecrement);
+	this->commandBuffer[4] = uint8_t(ssd1306::HardwareConfigCommands::DisplayStartLine);
+	this->commandBuffer[4] |= 0;		// Range 0-63
+	transaction_success &= RF_CALL(writeCommands(5));
+
+	this->commandBuffer[0] = uint8_t(ssd1306::HardwareConfigCommands::ComPinsOrder);
+	this->commandBuffer[1] = Height == 64 ? 0x12 : 0x02;
+	this->commandBuffer[2] = uint8_t(ssd1306::FundamentalCommands::ContrastControl);
+	this->commandBuffer[3] = 0xCF;	// Strange non-linear beahaviour
+	this->commandBuffer[4] = uint8_t(ssd1306::TimingAndDrivingCommands::PreChargePeriod);
+	this->commandBuffer[5] = 1;		// [3:0] Phase 1 period
+	this->commandBuffer[5] |= 15 << 4;// [7:4] Phase 2 period
+	transaction_success &= RF_CALL(writeCommands(6));
+
+	this->commandBuffer[0] = uint8_t(ssd1306::TimingAndDrivingCommands::V_DeselectLevel);
+	this->commandBuffer[1] = 4 << 4;	// [7:4] See Datasheet
+	this->commandBuffer[2] = uint8_t(ssd1306::ScrollingCommands::DisableScroll);
+	this->commandBuffer[3] = uint8_t(ssd1306::FundamentalCommands::EntireDisplayResumeToRam);
+	this->commandBuffer[4] = uint8_t(ssd1306::FundamentalCommands::NormalDisplay);
+	transaction_success &= RF_CALL(writeCommands(5));
+
+	this->commandBuffer[0] = uint8_t(ssd1306::FundamentalCommands::DisplayOn);
+	transaction_success &= RF_CALL(writeCommands(1));
+
+	RF_END_RETURN(transaction_success);
+}
+
+/**
+ * @brief	MemoryMode::HORIZONTAL and MemoryMode::VERTICAL
+ * 			have the best performance cause the whole buffer
+ * 			is send in one transaction.
+ */
+template<class I2cMaster, uint8_t Height>
+modm::ResumableResult<void>
+modm::Ssd1306I2c<I2cMaster, Height>::initializeMemoryMode()
+{
+	RF_BEGIN();
+	this->commandBuffer[0] = uint8_t(ssd1306::AdressingCommands::MemoryMode);
+	this->commandBuffer[1] = uint8_t(ssd1306::MemoryMode::HORIZONTAL);
+	transaction_success &= RF_CALL(writeCommands(2));
+
+	// Default on Power-up - can be omitted
+	this->commandBuffer[0] = uint8_t(ssd1306::AdressingCommands::ColumnAddress);
+	this->commandBuffer[1] = 0;
+	this->commandBuffer[2] = 127;
+	this->commandBuffer[3] = uint8_t(ssd1306::AdressingCommands::PageAddress);
+	this->commandBuffer[4] = 0;
+	this->commandBuffer[5] = 7;
+	transaction_success &= RF_CALL(writeCommands(6));
+
+	RF_END();
+}
+
+// ----------------------------------------------------------------------------
+template<class I2cMaster, uint8_t Height>
+modm::ResumableResult<void>
+modm::Ssd1306I2c<I2cMaster, Height>::startWriteDisplay()
+{
+	RF_BEGIN();
+
+	RF_WAIT_UNTIL(
+		this->transaction.configureDisplayWrite((uint8_t*)(&this->buffer), sizeof(this->buffer)) and
```

Comment:
`buffer` is a `uint8_t[][]`. This could be written without the C-style cast as `&this->buffer[0][0]`.